examples/common/python/connectors/ethereum/contracts/proxy-model/WorkOrderRegistry.sol

```diff
@@ -0,0 +1,171 @@
+/* Copyright 2019 iExec Blockchain Tech
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at 
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+pragma solidity ^0.5.0;
+
+import "./libs/SignatureVerifier.sol";
+
+contract WorkOrderRegistry is SignatureVerifier
+{
+	uint256 constant public TIMEOUT = 1 days;
+
+	enum WorkOrderStatus
+	{
+		NULL,
+		ACTIVE,
+		COMPLETED,
+		FAILED
+	}
+
+	struct WorkOrder
+	{
+		uint256 status;
+		uint256 timestamp;
+		bytes32 workerID;
+		bytes32 requesterID;
+		bytes   request;
+		address verifier;
+		uint256 returnCode;
+		bytes   response;
+	}
+
+	// workOrderID → workOrder
+	mapping(bytes32 => WorkOrder) m_workorders;
+
+	event workOrderSubmitted(
+		bytes32 indexed workOrderID,
+		bytes32 indexed workerID,
+		bytes32 indexed requesterID);
+
+	event workOrderCompleted(
+		bytes32 indexed workOrderID,
+		uint256 indexed workOrderReturnCode);
+
+	event workOrderTimedout(
+		bytes32 indexed workOrderID);
+
+	constructor()
+	public
+	{}
+
+	function workOrderSubmit(
+		bytes32       _workerID,
+		bytes32       _requesterID,
+		bytes memory _workOrderRequest,
+		address       _verifier,
+		bytes32       _salt)
```

Comment:
We have a PR on the specs that explains that

Reply:
Could you share the link for the PR for reference

Reply:
https://github.com/EntEthAlliance/trusted-computing/commit/8c0edb32c112ec4140e8d38b60d731b5f1da0f0b#diff-dd6b3e3eb82f946145f0bf2317833fa5

Reply:
I can't access this link. Seems it's a private repo

Reply:
Do you have the access to github of the TC spec: https://github.com/EntEthAlliance/trusted-computing

Reply:
No. Could you let me know how do I request access.

Reply:
Actually,  adding "salt" here is for the uniqueness

Reply:
you need to request eea admin

Reply:
Sure. I'll check with @EugeneYYY . Existing spec as `workOrderId` parameter in `workOrderSubmit` function. So earlier was `workOrderId` used for uniqueness ?

Reply:
The PR removes workOrderId has an input parameter. It is not clearly defined how this should be computed beforehand and it causes issues: what if the value provided is already used ? How to prevent race conditions ?

The solution (which is the PR) is for the requester to provide the parameters (and a random salt). The contract will then generate the workOrderId itself, which is very unlikely to collide.

Without the salt a colision would happen if a requester asks the same computation twice (same parameters) ... if we add a random salt generated on the requester's side we remove this issue.

Reply:
Why do we have a secret spec? Why cannot the spec be publicly accessible?